hooks/read-package-hook/test/createPeerDependencyPatcher.test.ts

```diff
@@ -114,3 +114,40 @@ test('createPeerDependencyPatcher() does not create duplicate extended ranges',
     nopadding: '15.0.1 || 16 || ^17.0.1 || 18.x',
   })
 })
+
+test('createPeerDependencyPatcher() overrides peerDependencies when parent>child selector is used', () => {
+  const patcher = createPeerDependencyPatcher({
+    allowedVersions: {
+      bar: '2',
+      'foo>bar': '0 || 1',
+    },
+  })
+  const patchedPkg = patcher({
+    peerDependencies: {
+      bar: '0 || 1',
+    },
+  }) as ProjectManifest
+  expect(patchedPkg.peerDependencies).toStrictEqual({
+    bar: '0 || 1 || 2',
+  })
+
+  // "foo" package is overridden correctly
+  const patchedFooPkg = patcher({
+    name: 'foo',
+    peerDependencies: {
+      bar: '0 || 1',
+    },
+  }) as ProjectManifest
+  expect(patchedFooPkg.peerDependencies).toStrictEqual({
+    bar: '0 || 1',
```

Comment:
how do you know if it was overriden correctly if the overriden range matches the original one. I guess use some other range at line 138

Reply:
Yeah I was trying to show with the previous test case, but I'll add a couple more test cases